### DISCUSSION TITLE
Globally extend the `env` function return type with Zod schema type

### DISCUSSION THREAD

**Discussion by jamiehaywood** [honojs/hono#1737] [General]
This is less of a Hono question, and more a TypeScript question, but I'm hoping someone can help me / tell me that this isn't possible.

# Problem
I have a Zod schema:
```ts
// src/env.ts
import { z } from "zod";

export const envSchema = z.object({
  FOO: z.string(),
  BAR: z.string(),
});

export type Env = z.infer<typeof envSchema>;
```

I want to globally override the return type of the `env()` function, so that I don't need to pass in a generic every time I invoke `env()`, i.e: 

```ts
// src/handler.ts
const app = new Hono();

app.get("/auth", async (c) => {
  // If possible I'd like to avoid doing - const { FOO } = env<Env>(c)
  const { FOO } = env(c); // would like FOO to be available with intellisense
})
```

# Attempted solution
I've tried globally extending the type in a `global.d.ts`, but I am having issues with the types playing nice:
```ts
// global.d.ts
import { Context } from "hono";
import { Env } from "./src/env";

declare module "hono" {
  interface Context<T = {}, U = any, V = {}> {
    env: Env & V;
  }
}
```

Any suggestions here would be greatly appreciated 🙏🏼

**Comment by jamiehaywood** [comment]
So a follow up to this, which achieves what I want. You can pass the Hono constructor a generic to the `Bindings` property:
```ts
const app = new Hono<{ Bindings: Env }>();
```

**Reply by yusukebe** [reply]
Yes. Please go in this way.
